src/main/java/com/google/devtools/build/lib/cmdline/RepositoryName.java

```diff
@@ -207,6 +207,13 @@ public static void validateUserProvidedRepoName(String name) throws EvalExceptio
     }
   }
 
+  /**
+   * Returns true if the given name cannot possibly be a canonical repository name.
+   */
+  public boolean isApparent() {
```

Comment:
IMO this method name is a bit misleading: all `RepositoryName` objects are (or _should_ be) canonical. Maybe make this a static method instead?